examples/miscellaneous/miscellaneous_ex1/miscellaneous_ex1.C

```diff
@@ -467,28 +467,16 @@ void assemble_wave(EquationSystems & es,
 
   // Note that we have not applied any boundary conditions so far.
   // Here we apply a unit load at the node located at (0,0,0).
-  {
-    // Iterate over local nodes
-    MeshBase::const_node_iterator           nd = mesh.local_nodes_begin();
-    const MeshBase::const_node_iterator nd_end = mesh.local_nodes_end();
-
-    for (; nd != nd_end; ++nd)
+  for (const auto & node : mesh.local_node_ptr_range())
+    if (std::abs(node(0)) < TOLERANCE &&
+        std::abs(node(1)) < TOLERANCE &&
+        std::abs(node(2)) < TOLERANCE)
```

Comment:
Wait, this can't be right.  You've changed node from a reference to a pointer, so we need to dereference it now before calling ```operator()()``` or ```dof_number()```.

Reply:
Why does `node(0)` even compile?

Reply:
It doesn't.  This is an ENABLE_INFINITE_ELEMENTS-only example.